src/VisualStudio/Core/Next/Remote/ServiceHubRemoteHostClient.cs

```diff
@@ -98,5 +102,19 @@ private void OnRpcDisconnected(object sender, JsonRpcDisconnectedEventArgs e)
         {
             Disconnected();
         }
+
+        private static async Task<Stream> RequestServiceAsync(HubClient client, string serviceName, string hostGroup, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            // we can remove these once whole system moved to new servicehub API
+            try
+            {
+                var descriptor = new ServiceDescriptor(serviceName) { HostGroup = new HostGroup(hostGroup) };
+                return await client.RequestServiceAsync(descriptor, cancellationToken).ConfigureAwait(false);
+            }
+            catch
```

Comment:
can we only catch the specific exception that gets thrown when the ServiceDescriptor route isn't avialable?